gossip/src/duplicate_shred_handler.rs

```diff
@@ -83,6 +104,13 @@ impl DuplicateShredHandler {
             return;
         }
         self.last_root = last_root;
+        self.enable_gossip_duplicate_proof_ingestion_epoch = {
+            let root_bank = self.bank_forks.read().unwrap().root_bank();
+            root_bank
+                .feature_set
+                .activated_slot(&feature_set::enable_gossip_duplicate_proof_ingestion::id())
+                .map(|slot| self.epoch_schedule.get_epoch(slot))
+        };
```

Comment:
Can we gate this behind a `enable_gossip_duplicate_proof_ingestion_epoch.is_none()` check to avoid taking the lock when possible?